CunnecryConverterBuildSprint1/CunnecryConverterBuildSprint1/ConverterViewController.swift

```diff
@@ -0,0 +1,115 @@
+    
+    @IBOutlet weak var currencyDisplay: UILabel!
+    
+    let currencyController = CurrencyController()
+    
+    var newRates: [String: Double] = [:]
+    
+    let currencyNames = ["CAD, Canadian Dollar",
+                         "HKD, Hong Kong Dollar",
+                         "ISK, Icelandic Króna",
+                         "PHP, Philippine peso",
+                         "DKK, Danish Krone",
+                         "HUF, Hungarian Forint",
+                         "CZK, Czech Koruna",
+                         "GBP, Pound sterling",
+                         "RON, Romanian Leu",
+                         "SEK, Swedish Krona",
+                         "IDR, Indonesian Rupiah",
+                         "INR, Indian Rupee",
+                         "BRL, Brazilian Real",
+                         "RUB, Russian Ruble",
+                         "HRK, Croatian Kuna",
+                         "JPY, Japanese Yen",
+                         "THB, Thailand Baht",
+                         "CHF, Swiss Franc",
+                         "EUR, Euro",
+                         "MYR, Malaysian Ringgit",
+                         "BGN, Bulgarian Lev",
+                         "TRY, Turkish lira",
+                         "CNY, Chinese Yuan",
+                         "NOK, Norwegian Krone",
+                         "NZD, New Zealand Dollar",
+                         "ZAR, South African Rand",
+                         "USD, United States Dollar",
+                         "MXN, Mexican Peso",
+                         "SGD, Singapore Dollar",
+                         "AUD, Australian Dollar",
+                         "ILS, Israeli New Shekel",
+                         "KRW, South Korean Won",
+                         "PLN, Poland złoty"]
+    
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return newRates.count
+    }
+    
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return "\(currencyNames.sorted()[row])"
+    }
+    
+    
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        
+        let currencyKey = Array(newRates.keys.sorted())[row]
+        guard let currency1 = newRates[currencyKey] else { return }
+        guard let currency2 = NumberFormatter().number(from: amountToConvertTextField.text!)?.doubleValue else { return }
+        
+        
+        func conversion() {
+            let result = currency1 * currency2
+            convertedAmountTextField.text = "\(formatter.string(for: result)!)"
+            currencyDisplay.text =   "\(newRates.keys.sorted()[row]) \(newRates[currencyKey]!)"
+            
+            
+        }
```

Comment:
Though you can nest functions inside other functions as you have done here, I wonder if it makes sense in this scenario, or if this function would be clearer on it's own.